apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -90,6 +99,118 @@ export function createDefaultSpriteMetadata(listData) {
   });
 }
 
+function generateAnimationMetadataForFile(fileObject) {
+  const json = fileObject.json;
+  const png = fileObject.png;
+  return getAnimationLibraryFile(json.key)
+    .then(metadata => {
+      // Metadata contains name, frameCount, frameSize, looping, frameDelay
+      let combinedMetadata = {
+        ...metadata,
+        jsonLastModified: json.last_modified,
+        pngLastModified: png.last_modified,
+        version: png.version_id,
+        sourceUrl: `/api/v1/animation-library/level_animations/${
+          png.version_id
+        }/${metadata.name}.png`,
+        sourceSize: png.source_size
+      };
+      return Promise.resolve(combinedMetadata);
+    })
+    .catch(err => {
+      return Promise.reject(err);
+    });
+}
+
+export function buildAnimationMetadata(files) {
+  let animationMetadataByName = {};
+  let resolvedPromisesArray = [];
+  for (const [fileKey, fileObject] of Object.entries(files)) {
+    resolvedPromisesArray.push(
+      generateAnimationMetadataForFile(fileObject).then(metadata => {
+        animationMetadataByName[fileKey] = metadata;
+      })
+    );
+  }
+  return Promise.all(resolvedPromisesArray).then(() => {
+    return animationMetadataByName;
+  });
+}
+
+function buildMap(
+  animationMetadata,
+  getStandardizedContent,
+  normalizingFunction
+) {
+  let contentMap = {};
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let formattedArray = getStandardizedContent(metadata);
+    formattedArray.map(item => {
+      let content = normalizingFunction ? normalizingFunction(item) : item;
+      // De-duplicate values as map is built
+      if (!contentMap[content]) {
+        contentMap[content] = new Set();
+      }
+      contentMap[content].add(key);
+    });
+  }
+
+  // After map is populated, transform values in sorted arrays
+  for (const key of Object.keys(contentMap)) {
+    contentMap[key] = [...contentMap[key]].sort();
+  }
+  return contentMap;
+}
+
+function getStandardizedAliases(metadata) {
+  const formattedAliases = metadata.aliases?.map(alias => {
+    alias.toLowerCase();
+  });
+  // Include the name of the animation as an alias
+  return [metadata.name.toLowerCase(), ...formattedAliases];
+}
+
+function getStandardizedCategories(metadata) {
+  // If the animation doesn't have a category, place it in a section for
+  // level-specific/hidden-from-library animations.
+  return metadata.categories ? metadata.categories : ['level_animations'];
```

Comment:
nit: a simpler way to express a fallback like this would be `metadata.categories || ['level_animations']`